src/auth/src/credentials/user_credential.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::credentials::traits::dynamic::Credential;
+use crate::credentials::Result;
+use crate::errors::{BoxError, CredentialError};
+use crate::token::{Token, TokenProvider};
+use http::header::{HeaderName, HeaderValue};
+
+/// Data model for a UserCredential
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+pub(crate) struct UserCredential {
+    token_provider: Box<dyn TokenProvider>,
+}
```

Comment:
Do you expect to dynamically change the type of token provider here?  If not, consider:

```suggestion
pub(crate) struct UserCredential<T> where T: impl TokenProvider + /* whatever else you need */ {
    token_provider: T,
}
```

You can still mock, you can still decorate this with tracing (with `T == TracingProvider<U>`), and you can avoid all the dynamic allocations.

Reply:
> Do you expect to dynamically change the type of token provider here?

No, I expect we will make a concrete implementation of `TokenProvider`, feed it to this `UserCredential` struct, and never change it.

> If not, consider...

Adopted.